src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,352 @@
+import { IOIcons, Icon } from "../icons";
+import { HSpacer } from "../spacer";
+import { LabelSmall } from "../typography";
+import { IconButton } from "../buttons";
+import { triggerHaptic } from "../../functions";
+
+type InputStatus = "initial" | "focused" | "disabled" | "error";
+
+interface InputTextProps {
+  placeholder: string;
+  value: string;
+  onChangeText: (value: string) => void;
+  icon?: IOIcons;
+  counterLimit?: number;
+  helperText?: string;
+  disabled?: boolean;
+  secretInput?: boolean;
+  onValidate?: (value: string) => boolean;
+  errorMessage?: string;
+}
+
+const styles = StyleSheet.create({
+  helperRow: {
+    ...IOStyles.row,
+    alignItems: "center",
+    paddingHorizontal: 10
+  },
+  textInput: {
+    ...IOStyles.row,
+    alignItems: "center",
+    paddingVertical: 8,
+    height: 60,
+    borderRadius: 8,
+    paddingHorizontal: 12
+  },
+  textInputStyle: {
+    fontSize: 16,
+    marginTop: IOSpacingScale[2],
+    fontWeight: "400",
+    lineHeight: 24,
+    fontFamily: "Readex Pro",
+    height: "100%"
+  },
+  textInputLabelWrapper: {
+    position: "absolute",
+    paddingHorizontal: 10,
+    zIndex: 10,
+    bottom: 0,
+    top: 0,
+    justifyContent: "center"
+  }
+});
+
+type InputTextHelperRow = {
+  value: string;
+  counterLimit?: number;
+  helperText?: string;
+  errorMessage?: string;
+  isValid?: boolean;
+};
+
+const HelperRow = ({
+  value,
+  counterLimit,
+  helperText,
+  errorMessage,
+  isValid
+}: InputTextHelperRow) => {
+  const valueCount = useMemo(() => value.length, [value]);
+
+  const helperRowStyle: ViewStyle = useMemo(() => {
+    if (counterLimit && helperText) {
+      return {
+        justifyContent: "space-between"
+      };
+    }
+    if (counterLimit) {
+      return {
+        justifyContent: "flex-end"
+      };
+    }
+    if (helperText) {
+      return {
+        justifyContent: "flex-start"
+      };
+    }
+    return {};
+  }, [counterLimit, helperText]);
+
+  return (
+    <View
+      style={[
+        IOStyles.row,
+        {
+          alignItems: "center",
+          paddingHorizontal: 10
+        },
+        helperRowStyle
+      ]}
+    >
+      {errorMessage && isValid === false && (
+        <LabelSmall weight="Regular" color="error-600">
+          {errorMessage}
+        </LabelSmall>
+      )}
+      {helperText && !(errorMessage && isValid === false) && (
+        <LabelSmall weight="Regular" color="grey-700">
+          {helperText}
+        </LabelSmall>
+      )}
+      {counterLimit && (
+        <LabelSmall
+          weight="Regular"
+          color="grey-700"
+        >{`${valueCount} / ${counterLimit}`}</LabelSmall>
+      )}
+    </View>
+  );
+};
+
+export const TextInputBase = ({
+  disabled = false,
+  placeholder,
+  value = "",
+  icon,
+  counterLimit,
+  helperText,
+  onChangeText,
+  onValidate,
+  secretInput,
+  errorMessage
+}: InputTextProps) => {
+  const labelSharedValue = useSharedValue(0);
```

Comment:
```suggestion
  const labelSharedValue = useSharedValue<boolean>(false);
```
If `labelSharedValue` acts as boolean, we should specify it and then make the animation simpler with a condition, without `interpolation` method:

```
fontSize: withTiming(labelSharedValue.value ? 12 : 16, {
      duration: 300,
      easing: Easing.elastic(0.85)
    }),
```